web/src/hotspots.ts

```diff
@@ -57,27 +60,34 @@ const HOTSPOT_LOCATIONS = new Map([
     ],
 ]);
 
-const meta = {
+const meta: {
+    opened_hotspot_name: null | string;
+} = {
     opened_hotspot_name: null,
 };
 
-function compute_placement($elt, popover_height, popover_width, prefer_vertical_positioning) {
-    const client_rect = $elt.get(0).getBoundingClientRect();
+function compute_placement(
+    $elt: JQuery,
```

Comment:
You can probably make this type more precise.

Reply:
The element don't have a specific property. It is button at one place, span at one place and div at 2 instances.

Reply:
`JQuery` is fine here (it defaults to `JQuery<HTMLElement>`).